java/code/src/com/suse/manager/xmlrpc/iss/HubHandler.java

```diff
@@ -388,15 +405,167 @@ public int deregister(User loggedInUser, String fqdn, boolean onlyLocal) {
      *  @apidoc.returntype #return_int_success()
      */
     public int setDetails(User loggedInUser, String fqdn, String role, Map<String, String> data) {
         ensureSatAdmin(loggedInUser);
-        hubManager.updateServerData(loggedInUser, fqdn, IssRole.valueOf(role), data);
+        ensureValidRole(role);
+        try {
+            hubManager.updateServerData(loggedInUser, fqdn, IssRole.valueOf(role), data);
+        }
+        catch (IllegalArgumentException ex) {
+            throw new InvalidParameterException(logGetErrorString(ex, "Illegal state"));
+        }
+        return 1;
+    }
+
+    private void ensureValidRole(String role) throws PermissionCheckFailureException {
+        try {
+            IssRole.valueOf(role);
+        }
+        catch (IllegalArgumentException ex) {
+            throw new InvalidParameterException(logGetErrorString(
+                    role, "Invalid role, must either be HUB or PERIPHERAL"));
+        }
+    }
+
+    /**
+     * Collect data about a Manager Server
+     *
+     * @param loggedInUser the user
+     * @return a {@link ManagerInfoJson} on success, exception otherwise
+     * @apidoc.doc Get manager info.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype $ManagerInfoJsonSerializer
+     */
+    @ReadOnly
+    public ManagerInfoJson getManagerInfo(User loggedInUser) {
+        ensureSatAdmin(loggedInUser);
+        try {
+            return hubManager.collectManagerInfo(loggedInUser);
+        }
+        catch (Exception ex) {
+            throw new InvalidParameterException(logGetErrorString(ex, "Error while collecting manager info"));
+        }
+    }
+
+    /**
+     * Collect data about all peripheral organizations
+     *
+     * @param loggedInUser the user
+     * @return a list of {@link Org} on success, exception otherwise
+     * @apidoc.doc Collect data about all peripheral organizations.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype
+     *      #return_array_begin()
+     *          $OrgSerializer
+     *      #array_end()
+     */
+    @ReadOnly
+    public List<Org> listAllPeripheralOrgs(User loggedInUser) {
+        ensureSatAdmin(loggedInUser);
+        try {
+            return hubManager.collectAllOrgs(loggedInUser);
+        }
+        catch (Exception ex) {
+            throw new InvalidParameterException(logGetErrorString(ex,
+                    "Error while collecting all peripheral organization"));
+        }
+    }
+
+    /**
+     * Collect data about all peripheral channels
+     *
+     * @param loggedInUser the user
+     * @return a list of {@link Channel} on success, exception otherwise
+     * @apidoc.doc Collect data about all peripheral channels.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype
+     *      #return_array_begin()
+     *          $ChannelSerializer
+     *      #array_end()
+     */
+    @ReadOnly
+    public List<Channel> listAllPeripheralChannels(User loggedInUser) {
```

Comment:
Also this is missleading as it sound like you would call it on a hub and it connect to a peripheral server to get the data.
But `hubManager.collectAllChannels()` just get the local channels.
I think we need to check if we need this at all for any workflow which makes sense to develop via API.